#!/usr/bin/env python
PACKAGE = "fav_control"

import roslib
roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dynamic_reconfigure", bool_t, 0, "Enable Changing Parameters", False)
gen.add("controller_type",    int_t,    0, "0=integral-SMC, 1=PID, 2=SMC + separate i", 0,  0, 2)
gen.add("reset_integrator", bool_t, 0, "Reset the Integrator to zero",    False)
gen.add("int_sat", double_t, 0, "Maximum integrated Error",    0.01, 0)
gen.add("k_u", double_t, 0, "Adjust gains due to uncertainty",    2.0, 0)

pid = gen.add_group('pid')
pid.add("k_p", double_t, 0, "Parameter PID",    9.0, 0)
pid.add("k_d", double_t, 0, "Parameter  PID",    5.8, 0)
pid.add("k_i", double_t, 0, "Parameter  PID",    0.2, 0)

smc = gen.add_group('smc')
smc.add("alpha", double_t, 0, "Parameter SMC",    1.5, 0)
smc.add("Lambda", double_t, 0, "Parameter SMC",    1.5, 0)
smc.add("kappa", double_t, 0, "Parameter SMC",    2.5, 0)
smc.add("epsilon", double_t, 0, "Parameter SMC",    0.4, 0)

exit(gen.generate(PACKAGE, "control", "PitchControl"))
